### ISSUE TITLE
Downloading large datasets fail and start over

### ISSUE THREAD

**Issue by bounlu** [ncbi/datasets#183]
I am trying to download large dataset of genomes with `datasets` but I experience very unstable and interrupted downloads.

**Command:**
```
taxon="bacteria"
datasets download genome taxon "$taxon" --filename "$taxon.zip" --dehydrated
unzip "$taxon.zip" -d "ncbi/$taxon"
rm "$taxon.zip"
datasets rehydrate --directory "ncbi/$taxon" --max-workers 20
```

**Error 1:**
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x8f08a1]

goroutine 15 [running]:
datasets_cli/v2/datasets.downloadFileWorker.func2(0xc000d09f48, 0xc000494000, 0xc000d09f30, 0x0?, 0x0?)
	apps/public/Datasets/v2/datasets/datasets/Rehydrate.go:190 +0x2c1
datasets_cli/v2/datasets.downloadFileWorker(0x0?, 0x0?, 0xc0000f4f00)
	apps/public/Datasets/v2/datasets/datasets/Rehydrate.go:217 +0xf7
created by datasets_cli/v2/datasets.downloadMultipleFiles
	apps/public/Datasets/v2/datasets/datasets/Rehydrate.go:245 +0x14b
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x8f08a1]
```

**Error 2:**
`Error:  http2: server sent GOAWAY and closed the connection; LastStreamID=4097, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed"`

**Error 3:**
`Error:  unexpected EOF unexpected EOF unexpected EOF unexpected EOF unexpected EOF unexpected EOF unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=3, ErrCode=NO_ERROR, debug="timeout" unexpected EOF unexpected EOF unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=167, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=31, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=11, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=19, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=11, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=27, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=33, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=29, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" unexpected EOF unexpected EOF unexpected EOF unexpected EOF unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=39, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=39, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=39, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" http2: server sent GOAWAY and closed the connection; LastStreamID=1, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" http2: server sent GOAWAY and closed the connection; LastStreamID=9, ErrCode=NO_ERROR, debug="timeout" unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" http2: server sent GOAWAY and closed the connection; LastStreamID=149, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=23, ErrCode=NO_ERROR, debug="timeout" unexpected EOF unexpected EOF unexpected EOF unexpected EOF unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=95, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=31, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=31, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=31, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=31, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=31, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=11, ErrCode=NO_ERROR, debug="timeout" unexpected EOF unexpected EOF unexpected EOF unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=29, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=17, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=25, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=5, ErrCode=NO_ERROR, debug="timeout" unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=27, ErrCode=NO_ERROR, debug="timeout" unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" unexpected EOF unexpected EOF unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=11, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug="" unexpected EOF unexpected EOF unexpected EOF unexpected EOF unexpected EOF unexpected EOF unexpected EOF unexpected EOF unexpected EOF unexpected EOF unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=9, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=11, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=11, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=11, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=11, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=21, ErrCode=NO_ERROR, debug="timeout" unexpected EOF unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=25, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=25, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=25, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=25, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=25, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=11, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=7, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=7, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=11, ErrCode=NO_ERROR, debug="timeout" unexpected EOF unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=25, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=23, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=37, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=5, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=3, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=17, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=203, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=39, ErrCode=NO_ERROR, debug="timeout" http2: server sent GOAWAY and closed the connection; LastStreamID=55, ErrCode=NO_ERROR, debug="timeout" unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=179, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=179, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=179, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=179, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=179, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=179, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=179, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=179, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=179, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed" http2: server sent GOAWAY and closed the connection; LastStreamID=41, ErrCode=NO_ERROR, debug="timeout" unexpected EOF http2: server sent GOAWAY and closed the connection; LastStreamID=83, ErrCode=NO_ERROR, debug="timeout"`

The worst thing is when I try to rerun `rehydrate` to resume the download, it starts over and downloads everything from the beginning and overwrites. This cycle repeats after about 1 day when the download is interrupted again.

Is there a way to fix this?

**Comment by ericcox1** [comment]
Hi Ömer,

Thanks for the detailed error report. We hope to release a fix soon that will include retries and restore the ability to resume downloads. I will comment here and on issue #83 when the fix is ready.

Thanks,
Eric

Eric Cox, PhD [Contractor] (he/him/his)
NCBI Datasets
Sequence Enhancements, Tools and Delivery (SeqPlus)
NIH/NLM/NCBI
[eric.cox@nih.gov](mailto:eric.cox@nih.gov)

**Comment by bounlu** [comment]
Hi Eric,

Thanks for working on the fix. It will be great to have the release as soon as possible as I need to download data to proceed with my analysis.

Thanks.

**Comment by ericcox1** [comment]
Hi Ömer,

We have released a fix to production. Please upgrade to `14.2.1` and try again.

If you don't mind, could you please share your results here?

Best,
Eric

**Comment by bounlu** [comment]
Hi Eric,

I downloaded and tested the new version. It seems resume feature works fine now:

```
for taxon in "Salmonella enterica" "Listeria monocytogenes" "Escherichia coli" "Staphylococcus aureus"
do
   taxon2=$(echo "$taxon" | tr " " "_") && \
   until datasets rehydrate --directory "ncbi/$taxon2" --max-workers 20
   do
      :
   done
done
Found 453252 files for rehydration
Completed 2867 of 384754 [------------------------------------------------]   1%
^C
```

Many thanks for the quick fix and help.
Best.

**Comment by ganiatgithub** [comment]
Hi Eric,

I seem to have the same issue with `datasets version: 14.26.0`. Here are my error messages with two trials I had:

```
Collecting 1,520  records [================================================] 100% 1520/1520
Downloading: zou_2019.zip    593kB done
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x923899]

goroutine 11 [running]:
datasets_cli/v2/datasets.baseRetryPolicy(...)
        apps/public/Datasets/v2/datasets/datasets/root.go:225
datasets_cli/v2/datasets.DefaultRetryPolicy({0xb91b10, 0xc00003c138}, 0x0, {0xc0003ba060?, 0x4e24c7?})
        apps/public/Datasets/v2/datasets/datasets/root.go:195 +0x59
github.com/hashicorp/go-retryablehttp.(*Client).Do(0xc0000eb880, 0xc0005707c8)
        external/com_github_hashicorp_go_retryablehttp/client.go:631 +0x61e
github.com/hashicorp/go-retryablehttp.(*RoundTripper).RoundTrip(0xc00000cfa8, 0xc000593bc0?)
        external/com_github_hashicorp_go_retryablehttp/roundtripper.go:44 +0x7d
datasets_cli/v2/datasets.LoggingRoundTripper.RoundTrip({{0xb8bdc0?, 0xc00000cfa8?}}, 0xc0001c0400)
        apps/public/Datasets/v2/datasets/datasets/root.go:522 +0x21d
net/http.send(0xc0001c0400, {0xb8e580, 0xc0001ac550}, {0xa7b3a0?, 0x1?, 0x0?})
        GOROOT/src/net/http/client.go:252 +0x5d8
net/http.(*Client).send(0xc00039c8a0, 0xc0001c0400, {0x480e49?, 0x630000c000206f06?, 0x0?})
        GOROOT/src/net/http/client.go:176 +0x9b
net/http.(*Client).do(0xc00039c8a0, 0xc0001c0400)
        GOROOT/src/net/http/client.go:725 +0x8f5
net/http.(*Client).Do(...)
        GOROOT/src/net/http/client.go:593
datasets_cli/v2/datasets.processHTTPRequest(0xc00042de50?, 0xc000593bc0?)
        apps/public/Datasets/v2/datasets/datasets/Rehydrate.go:150 +0xb9
datasets_cli/v2/datasets.downloadFileWorker.func2(0xc00003d290, 0xc00042df38, 0xc000424a20, 0xc00042df20, 0x0?, 0x0?, 0x0?)
        apps/public/Datasets/v2/datasets/datasets/Rehydrate.go:203 +0x3da
datasets_cli/v2/datasets.downloadFileWorker(0x0?, 0x0?, 0x0?, 0xc00007ad20, 0x0?)
        apps/public/Datasets/v2/datasets/datasets/Rehydrate.go:258 +0x110
created by datasets_cli/v2/datasets.downloadMultipleFiles
        apps/public/Datasets/v2/datasets/datasets/Rehydrate.go:319 +0x445
```

```
Collecting 737  records [================================================] 100% 737/737
Downloading: omalley_2021.zip    233kB done
Error: 
File unavailable ([gateway] 502 Bad Gateway): omalley_2021/ncbi_dataset/data/GCA_015062285.1/GCA_015062285.1_ASM1506228v1_genomic.fna
unexpected EOF
http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""
http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""
http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""
http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""
http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""
http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""
http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""
http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""
File unavailable ([gateway] 504 Gateway Timeout): omalley_2021/ncbi_dataset/data/GCA_015061365.1/GCA_015061365.1_ASM1506136v1_genomic.fna
http2: server sent GOAWAY and closed the connection; LastStreamID=189, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed"
http2: server sent GOAWAY and closed the connection; LastStreamID=189, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed"
http2: server sent GOAWAY and closed the connection; LastStreamID=189, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed"
http2: server sent GOAWAY and closed the connection; LastStreamID=189, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed"
http2: server sent GOAWAY and closed the connection; LastStreamID=189, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed"
http2: server sent GOAWAY and closed the connection; LastStreamID=189, ErrCode=unknown error code 0xfffffc7a, debug="The user callback function failed"
http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""
http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""
http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""
unexpected EOF
unexpected EOF
unexpected EOF
unexpected EOF
unexpected EOF
unexpected EOF
unexpected EOF
unexpected EOF
unexpected EOF
unexpected EOF
unexpected EOF
http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""
unexpected EOF
unexpected EOF
unexpected EOF
unexpected EOF
unexpected EOF
```

Could you take a look?

**Comment by mikeph52** [comment]
Greetings, I also experience the same problem in [datasets version 18.21.0](https://github.com/ncbi/datasets/releases/tag/v18.21.0), running from a conda environment. 
### The command:
```bash
datasets download genome taxon 6656  --reference --filename "$DB"/arthropoda.zip
datasets download genome taxon 1224  --reference --filename "$DB"/proteobacteria.zip
datasets download genome taxon 5878  --reference --filename "$DB"/ciliophora.zip
datasets download genome taxon 4890  --reference --filename "$DB"/ascomycota.zip
datasets download genome taxon 33090 --reference --filename "$DB"/streptophyta.zip
datasets download genome taxon 9606  --reference --filename "$DB"/human.zip
datasets download genome taxon 6073  --reference --filename "$DB"/cnidaria.zip
```
### The error log:
```
Error: Download error: http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""

Use datasets download genome taxon <command> --help for detailed help about a command.
```
The download fails only on large datasets, like the _Arthropoda_. 
Thank you

**Comment by mtntsuchiya** [comment]
Hi @mikeph52 ,
Thank you so much for reaching out. 

The arthropoda dataset is very large. In those cases, we recommend users to download a dehydrated package. You can find more info here. [https://www.ncbi.nlm.nih.gov/datasets/docs/v2/how-tos/genomes/large-download/](https://www.ncbi.nlm.nih.gov/datasets/docs/v2/how-tos/genomes/large-download/) .

For your case, the commands would be:
 
```
# Download a dehydrate package
datasets download genome taxon 6656  --reference --filename arthropoda-dh.zip --dehydrated

# Unzip
unzip arthropoda-dh.zip -d arthropoda

# Rehydrate
datasets rehydrate --directory arthropoda --max-workers 20
```

Please feel free to reach out if you have any other issues.

Best,
Mirian

**Comment by mikeph52** [comment]
Thanks for your help!!

**Comment by daddabi** [comment]
Hi, i'm facing similar issues though I'm using datasets download virus genome taxon. Since there is no --dehydrated option for virus genome command, do you suggest using datasets download genome taxon <taxid> --dehydrated instead? will this cause any issue?

**Comment by mtntsuchiya** [comment]
Hi @daddabi ,

Which virus taxon are you interested in downloading? 

The `genome` service (`datasets download genome`) retrieves sequences submitted specifically as genome assemblies; in the case of viruses, it corresponds to a subset of virus genomes that submitted as genomes assemblies or have been designated RefSeq records which are reference sequences for a particular species. The `virus` service (`datasets download virus genome`), by contrast, returns all virus sequences in the Nucleotide database. You can see the difference in numbers by looking at the viral genome assemblies [here](https://www.ncbi.nlm.nih.gov/datasets/genome/?taxon=10239) vs the virus service [here](https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/virus?VirusLineage_ss=taxid:10239&SeqType_s=Nucleotide).

If you could try to run the command again using the `--debug` flag and share the results here, I can pass the info to our developers to try to understand what's going on.

**Comment by daddabi** [comment]
> Hi [@daddabi](https://github.com/daddabi) ,
> 
> Which virus taxon are you interested in downloading?
> 
> The `genome` service (`datasets download genome`) retrieves sequences submitted specifically as genome assemblies; in the case of viruses, it corresponds to a subset of virus genomes that submitted as genomes assemblies or have been designated RefSeq records which are reference sequences for a particular species. The `virus` service (`datasets download virus genome`), by contrast, returns all virus sequences in the Nucleotide database. You can see the difference in numbers by looking at the viral genome assemblies [here](https://www.ncbi.nlm.nih.gov/datasets/genome/?taxon=10239) vs the virus service [here](https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/virus?VirusLineage_ss=taxid:10239&SeqType_s=Nucleotide).
> 
> If you could try to run the command again using the `--debug` flag and share the results here, I can pass the info to our developers to try to understand what's going on.

Hi, thanks for the clarification, that makes sense now. I didn’t realize the difference between the genome and virus services.
In this case, I’m interested in downloading all sequences from the Nucleotide database. I initially tried downloading a virus taxon for Influenza A (which contains ~2 million sequences), but the download failed. So I tested another taxon, Respiratory Syncytial Virus (RSV), which has around ~70k sequences, and encountered a similar issue.
Here is the debug output from that run:
```
datasets download virus genome taxon 11250 --filename rsv.zip --debug
2026/04/09 15:21:26
POST /datasets/v2/taxonomy/taxon_suggest HTTP/1.1
Host: api.ncbi.nlm.nih.gov
User-Agent: OpenAPI-Generator/1.0.0/go
Content-Length: 128
Accept: application/json
Content-Type: application/json
Ncbi-Phid: BD6C7DF730C33D8D0AF10D58
X-Datasets-Client: datasets-cli
X-Datasets-Client-Arch: amd64
X-Datasets-Client-Cmd: download virus genome taxon 11250 --filename rsv.zip --debug
X-Datasets-Client-Os: linux
X-Datasets-Client-Version: 18.22.1
Accept-Encoding: gzip

{"exact_match":true,"tax_rank_filter":"higher_taxon","taxon_query":"11250","taxon_resource_filter":"TAXON_RESOURCE_FILTER_ALL"}

2026/04/09 15:21:27
HTTP/2.0 200 OK
Access-Control-Expose-Headers: X-RateLimit-Limit,X-RateLimit-Remaining
Content-Security-Policy: upgrade-insecure-requests
Content-Type: application/json
Date: Thu, 09 Apr 2026 07:21:27 GMT
Grpc-Metadata-Via: h2 linkerd
Ncbi-Phid: BD6C7DF730C33D8D0AF10D58.1.1.1
Server: Finatra
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
X-Datasets-Version: 18.23.0
X-Datasets-Version-Message: New version of client (18.23.0) available at https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets.
X-Ratelimit-Limit: 5
X-Ratelimit-Remaining: 4
X-Ua-Compatible: IE=Edge
X-Xss-Protection: 1; mode=block


2026/04/09 15:21:27
POST /datasets/v2/taxonomy/dataset_report HTTP/1.1
Host: api.ncbi.nlm.nih.gov
User-Agent: OpenAPI-Generator/1.0.0/go
Content-Length: 51
Accept: application/json
Content-Type: application/json
Ncbi-Phid: BD6C7DF730C33D8D0AF10D58
X-Datasets-Client: datasets-cli
X-Datasets-Client-Arch: amd64
X-Datasets-Client-Cmd: download virus genome taxon 11250 --filename rsv.zip --debug
X-Datasets-Client-Os: linux
X-Datasets-Client-Version: 18.22.1
Accept-Encoding: gzip

{"returned_content":"METADATA","taxons":["11250"]}

2026/04/09 15:21:27
HTTP/2.0 200 OK
Access-Control-Expose-Headers: X-RateLimit-Limit,X-RateLimit-Remaining
Content-Security-Policy: upgrade-insecure-requests
Content-Type: application/json
Date: Thu, 09 Apr 2026 07:21:26 GMT
Grpc-Metadata-Via: h2 linkerd
Ncbi-Phid: BD6C7DF730C33D8D0AF10D58.2.1.1
Server: Finatra
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
X-Datasets-Version: 18.23.0
X-Datasets-Version-Message: New version of client (18.23.0) available at https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets.
X-Ncbi-Total-Count: 1
X-Ratelimit-Limit: 5
X-Ratelimit-Remaining: 3
X-Ua-Compatible: IE=Edge
X-Xss-Protection: 1; mode=block


2026/04/09 15:21:27
POST /datasets/v2/taxonomy/dataset_report HTTP/1.1
Host: api.ncbi.nlm.nih.gov
User-Agent: OpenAPI-Generator/1.0.0/go
Content-Length: 51
Accept: application/json
Content-Type: application/json
Ncbi-Phid: BD6C7DF730C33D8D0AF10D58
X-Datasets-Client: datasets-cli
X-Datasets-Client-Arch: amd64
X-Datasets-Client-Cmd: download virus genome taxon 11250 --filename rsv.zip --debug
X-Datasets-Client-Os: linux
X-Datasets-Client-Version: 18.22.1
Accept-Encoding: gzip

{"returned_content":"METADATA","taxons":["11250"]}

2026/04/09 15:21:28
HTTP/2.0 200 OK
Access-Control-Expose-Headers: X-RateLimit-Limit,X-RateLimit-Remaining
Content-Security-Policy: upgrade-insecure-requests
Content-Type: application/json
Date: Thu, 09 Apr 2026 07:21:28 GMT
Grpc-Metadata-Via: h2 linkerd
Ncbi-Phid: BD6C7DF730C33D8D0AF10D58.3.1.1
Server: Finatra
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
X-Datasets-Version: 18.23.0
X-Datasets-Version-Message: New version of client (18.23.0) available at https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets.
X-Ncbi-Total-Count: 1
X-Ratelimit-Limit: 5
X-Ratelimit-Remaining: 2
X-Ua-Compatible: IE=Edge
X-Xss-Protection: 1; mode=block


New version of client (18.23.0) available at https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets.
2026/04/09 15:21:28
POST /datasets/v2/virus/genome/download HTTP/1.1
Host: api.ncbi.nlm.nih.gov
User-Agent: OpenAPI-Generator/1.0.0/go
Content-Length: 282
Accept: text/plain,application/zip
Content-Type: application/json
Ncbi-Phid: BD6C7DF730C33D8D0AF10D58
X-Datasets-Client: datasets-cli
X-Datasets-Client-Arch: amd64
X-Datasets-Client-Cmd: download virus genome taxon 11250 --filename rsv.zip --debug
X-Datasets-Client-Os: linux
X-Datasets-Client-Version: 18.22.1
Accept-Encoding: gzip

{"annotated_only":false,"complete_only":false,"format":"tsv","geo_location":"","host":"","include_sequence":["GENOME"],"pangolin_classification":"","refseq_only":false,"released_since":"0001-01-01T00:00:00Z","taxons":["11250"],"updated_since":"0001-01-01T00:00:00Z","usa_state":""}

2026/04/09 15:21:28
HTTP/2.0 200 OK
Access-Control-Expose-Headers: X-RateLimit-Limit,X-RateLimit-Remaining
Content-Disposition: attachment; filename=ncbi_dataset.zip
Content-Security-Policy: upgrade-insecure-requests
Content-Type: application/zip
Date: Thu, 09 Apr 2026 07:21:27 GMT
Grpc-Metadata-Logging-Activity: download
Grpc-Metadata-Logging-Annotated_only: False
Grpc-Metadata-Logging-Refseq_only: False
Grpc-Metadata-Logging-Service: virus
Grpc-Metadata-Logging-Taxons: 11250
Grpc-Metadata-Logging-Taxons_count: 1
Grpc-Metadata-Via: h2 linkerd
Ncbi-Phid: BD6C7DF730C33D8D0AF10D58.4.1.1
Server: Finatra
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
Vary: Accept-Encoding
X-Datasets-Version: 18.23.0
X-Datasets-Version-Message: New version of client (18.23.0) available at https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets.
X-Ratelimit-Limit: 5
X-Ratelimit-Remaining: 2
X-Ua-Compatible: IE=Edge
X-Xss-Protection: 1; mode=block


Downloading: rsv.zip    130MB error
Error: Download error: http2: server sent GOAWAY and closed the connection; LastStreamID=2147483647, ErrCode=NO_ERROR, debug=""

Use datasets download virus genome taxon <command> --help for detailed help about a command.
```
